crates/diff_confused_web/minifier/main.ts

```diff
@@ -0,0 +1,124 @@
+                    await htmlMinifier(origin.toString(), {
+                        collapseBooleanAttributes: true,
+                        collapseWhitespace: true,
+                        decodeEntities: true,
+                        html5: true,
+                        minifyCSS: true,
+                        minifyJS: terserOptions,
+                        processConditionalComments: true,
+                        removeAttributeQuotes: true,
+                        removeComments: true,
+                        removeEmptyAttributes: true,
+                        removeOptionalTags: true,
+                        removeRedundantAttributes: true,
+                        removeScriptTypeAttributes: true,
+                        removeStyleLinkTypeAttributes: true,
+                        removeTagWhitespace: true,
+                        sortAttributes: true,
+                        sortClassName: true,
+                        trimCustomFragments: true,
+                        useShortDoctype: true,
+                    }),
+                );
+                break;
+            }
+
+            case ".css": {
+                minified = Buffer.from(
+                    new CleanCSS().minify(origin.toString()).styles,
+                );
+                break;
+            }
+
+            case ".js": {
+                const m = await jsMinifier(origin.toString(), terserOptions);
+                if (m.code == null) {
+                    throw new Error("terser returned nullish code");
+                }
+                minified = Buffer.from(m.code);
+                break;
+            }
+
+            default: {
+                compressed = false;
+                minified = origin;
+            }
+        }
+
+        await fs.writeFile(path.join(MINIFIED_DIR, originFileName), minified);
+        processedFiles.push(originFileName);
+    }
+
+    await showResult(processedFiles);
+}
+
+async function showResult(filenames: Array<string>) {
+    const filenameMaxLength = filenames
+        .map((x) => x.length)
+        .reduce((a, b) => Math.max(a, b));
+
+    console.log(
+        printf(
+            `%${filenameMaxLength}s: %9s %9s %9s`,
+            "filename",
+            "origin",
+            "minify",
+            "brotli",
+        ),
+    );
+
+    for (const filename of filenames) {
+        const kib = (hasLength: { length: number }) =>
+            printf("%6.02fKiB", hasLength.length / 1024);
+        const origin = await fs.readFile(path.join(ORIGINAL_DIR, filename));
+        const minified = await fs.readFile(path.join(MINIFIED_DIR, filename));
+        const compressed = brotli.compress(minified);
+
+        console.log(
+            printf(
+                `%${filenameMaxLength}s: %s %s %s`,
+                filename,
+                kib(origin),
+                kib(minified),
+                kib(compressed),
+            ),
+        );
+    }
+}
+
+main().catch(console.error);
```

Comment:
めちゃくちゃ細かいけど、catch してしまうとただ stderr にエラーが吐かれるだけで終了コードが 0 にならずおっかなかったので、できれば対策したいなあ (もはや catch しないで esrun にすべてを任せるとかでもいいとはちょっと思っている)

```suggestion
main();
```

Reply:
確かに。終了コードが常に正常扱いになるのは避けたいですね。修正しました。